x/stakeibc/keeper/msg_server_instant_redeem_stake_test.go

```diff
@@ -0,0 +1,201 @@
+package keeper_test
+
+import (
+	sdkmath "cosmossdk.io/math"
+	"fmt"
+	epochtypes "github.com/Stride-Labs/stride/v5/x/epochs/types"
+	recordtypes "github.com/Stride-Labs/stride/v5/x/records/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v5/x/stakeibc/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/stretchr/testify/suite"
+)
+
+type InstantRedeemStakeState struct {
+	epochNumber         uint64
+	depositRecordAmount sdkmath.Int
+	initialAmount       sdkmath.Int
+	hostZone            stakeibctypes.HostZone
+}
+
+type InstantRedeemStakeTestCase struct {
+	user         Account
+	zoneAccount  Account
+	initialState InstantRedeemStakeState
+	validMsg     stakeibctypes.MsgInstantRedeemStake
+}
+
+func (s *KeeperTestSuite) SetupInstantRedeemStake() InstantRedeemStakeTestCase {
+	unbondAmount := sdkmath.NewInt(1_000_000)
+	depositAmount := sdkmath.NewInt(500_000)
+	initialAmount := sdkmath.NewInt(1_000_000)
+	user := Account{
+		acc:           s.TestAccs[0],
+		atomBalance:   sdk.NewInt64Coin(IbcAtom, initialAmount.Int64()),
+		stAtomBalance: sdk.NewInt64Coin(StAtom, unbondAmount.Int64()),
+	}
+	s.FundAccount(user.acc, user.atomBalance)
+	s.FundAccount(user.acc, user.stAtomBalance)
+
+	zoneAddress := stakeibctypes.NewZoneAddress(HostChainId)
+
+	zoneAccount := Account{
+		acc:           zoneAddress,
+		atomBalance:   sdk.NewInt64Coin(IbcAtom, 10_000_000),
+		stAtomBalance: sdk.NewInt64Coin(StAtom, 10_000_000),
+	}
+	s.FundAccount(zoneAccount.acc, zoneAccount.atomBalance)
+	s.FundAccount(zoneAccount.acc, zoneAccount.stAtomBalance)
+
+	hostZone := stakeibctypes.HostZone{
+		ChainId:        HostChainId,
+		HostDenom:      Atom,
+		IbcDenom:       IbcAtom,
+		RedemptionRate: sdk.NewDec(1.0),
+		Address:        zoneAddress.String(),
+		StakedBal:      unbondAmount,
+	}
+
+	epochTracker := stakeibctypes.EpochTracker{
```

Comment:
I don't think you need the epoch tracker for these tests

Reply:
Yeah, I did when I wrote it initially, but I don't anymore

Reply:
@sampocs this is addressed. ready to resolve